code/surface_area/compute_surface_area.py

```diff
@@ -0,0 +1,172 @@
+from utils import find_surface_files, get_map_info
+
+my_runner = DockerRunner()
+my_runner.data_dir = Path(__file__).parent / "../../share/Inputs"
+
+
+def compute_surface_area(input_gifti: Path, output_metric: Path) -> None:
+    """
+    Compute the surface area of a brain surface mesh using niwrap.
+
+    """
+    # test if file exists
+    if not input_gifti.exists():
+        print(f"✗ Input file does not exist: {input_gifti}")
+        return
+
+    wb.surface_vertex_areas(surface=str(input_gifti), metric=str(output_metric))
+    print(f"Surface area metric saved to {output_metric}")
+
+
+def process_files(
+    input_files: Iterable[Path], validate_output: bool = False, dry_run: bool = False
+):
+    """
+    Process all input files to compute surface areas.
+    """
+    successful = 0
+    failed = 0
+
+    for input_gifti in input_files:
+        map_info = get_map_info(input_gifti)
+
+        new_name = f"tpl-{map_info['Space']}_den-{map_info['Density']}_hemi-{map_info['Hemi']}_desc-vaavg_midthickness.shape.gii"
+
+        # Generate output filename
+        output_metric = input_gifti.parent / new_name
+
+        print(f"\nProcessing: {input_gifti.name}")
+        print(f"Input: {input_gifti}")
+        print(f"Output: {output_metric}")
+
+        if dry_run:
+            print("  (DRY RUN - no processing performed)")
+            continue
+
+        try:
+            # Run the actual processing
+            compute_surface_area(input_gifti, output_metric)
+
+            # Test output file if requested
+            if validate_output:
+                if validate_output_file_data(output_metric):
+                    successful += 1
+                else:
+                    failed += 1
+                    print(f"✗ Validation failed for {output_metric}")
+            else:
+                successful += 1
+
+        except Exception as e:
+            print(f"✗ Failed to process {input_gifti}: {e}")
+            failed += 1
+
+    return successful, failed
+
+
+def main():
+    """
+    Main command-line interface function.
+    """
+    parser = argparse.ArgumentParser(
+        description="Compute surface area metrics for brain surface meshes",
+        formatter_class=argparse.RawDescriptionHelpFormatter,
+        epilog="""
+Examples:
+  python surface_area.py                              # Use default ../share/Inputs
+  python surface_area.py -i /path/to/surfaces         # Use absolute path
+  python surface_area.py -i ./data/surfaces           # Use relative path
+  python surface_area.py -i ../inputs --validate      # Validate outputs
+  python surface_area.py -i ../inputs --dry-run       # Show what would be processed
+  python surface_area.py -p "*mid*" "*thickness*"     # Custom patterns
+        """,
+    )
+    SCRIPT_DIR = Path(__file__).parent
+    parser.add_argument(
+        "-i",
+        "--input-dir",
+        type=Path,
+        default=Path(SCRIPT_DIR / "../../share/Inputs").resolve(),
```

Comment:
```suggestion
        default=(SCRIPT_DIR / "../../share/Inputs").resolve(),
```

Very minor thing here and doesn't really make a difference - don't need to include `Path` here. The resulting object from `SCRIPT_DIR / "../../share/Inputs"` is already a `Path`. For larger libraries, I would probably not include a default for something like this which is likely to be a required argument.